.github/workflows/ci.yml

```diff
@@ -0,0 +1,49 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  tests:
+    name: Tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: Swatinem/rust-cache@23bce251a8cd2ffc3c1075eaa2367cf899916d84
+      - uses: hrishikesh-kadam/setup-lcov@v1
```

Comment:
It is rather a good practice to always use hashes instead of tags for non-official actions.
https://blog.rafaelgss.dev/why-you-should-pin-actions-by-commit-hash :+1: